docs/orchestrating-elastic-stack-applications/beat.asciidoc

```diff
@@ -0,0 +1,378 @@
+
+[id="{p}-beat-deploy-community-beat"]
+=== Deploy a Community Beat
+
+ECK supports the deployment of any Community Beat. `type` and `version` specification elements have to be provided. In addition:
+
+- `image` must point to the image to be deployed
+-  a role must exist in Elasticsearch and have the permissions required by the Beat. The role name has to be `eck_beat_type_role`, where `type` is the Beat type. For example, when deploying `type: kafkabeat`, the role name would be `eck_beat_kafkabeat_role`.
+
+
+[id="{p}-beat-set-up-kibana-dashboards"]
+=== Set up Kibana dashboards
+
+ECK can instruct Beats to set up example dashboards packaged with the Beat. To enable this, set the `kibanaRef` element in the specification to point to ECK-managed Kibana deployment:
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: quickstart
+spec:
+  kibanaRef:
+    name: quickstart
+...
+----
+
+ECK will create a new user in Elasticsearch with a minimal set of appropriate roles and permissions that is needed for dashboard setup.
+
+
+[id="{p}-beat-secrets-keystore-for-secure-settings"]
+=== Secrets keystore for secure settings
+
+Beats offer a secret keystore for sensitive settings that need to be provided in the config, like for example passwords. This avoids storing them in the config directly.
+
+ECK exposes that mechanism with the `secureSettings` element in the specification. <<{p}-es-secure-settings,Similar to Elasticsearch>>, you can use Kubernetes Secrets to provide the settings in a secure way:
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: quickstart
+spec:
+  secureSettings:
+  - secretName: agent-name-secret
+  config:
+    name: ${AGENT_NAME_VAR}
+...
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: agent-name-secret
+stringData:
+  AGENT_NAME_VAR: id_007
+----
+
+See Beats link:https://www.elastic.co/guide/en/beats/filebeat/current/keystore.html[documentation] for more details.
+
+
+[id="{p}-beat-set-beat-output"]
+=== Set Beat output
+
+If the `elasticsearchRef` element is specified, ECK populates the output section of the Beat config. ECK creates a user with appropriate roles and permissions and uses its credentials. If required, it also mounts the CA certificate in all Beat Pods, and recreates Pods when this certificate changes.
+
+Output can be set to any that's supported by a given Beat. To do that, remove the `elasticsearchRef` element from the specification and include an appropriate output configuration in the `config` or `configRef`.
+
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: quickstart
+spec:
+  config:
+    output.redis:
+      hosts: ["my_redis.default.svc"]
+      password: "my_password"
```

Comment:
I think we probably do not want to encourage putting passwords in the config directly and should have an example here using configRef (and maybe even removing the text about it being possible to place in the config)

Reply:
Should I just change it to a config that doesn't contain any credentials? I worry that using `configRef` with a Secret will be confusing and removing `config` will imply that setting custom output is only possible via `configRef`.

Reply:
Well a custom output that requires credentials should not be in the `config` and should only be in the `configRef`. I'm good with whatever as long as we don't promote storing passwords in plaintext

Reply:
This can be true even when using `config` if user relies on env variables or Beats keystore. I've change it to kafka config that doesn't need to contain creds.